### PR TITLE
Add ProblemDetails helpers to Results

### LINKED ISSUE #34169: Add ProblemDetails helpers to minimal actions
* [ ] Move ProblemDetails from MVC lower in to the ASP.NET Core stack (type forward)
* [ ] Add ProblemDetails as a helper method.

ValidationProblemDetails is tricky because it references `ModelStateDictionary`, so we'll leave it be in MVC for the immediate future.

========

API Review:

* Type forward `Microsoft.AspNetCore.Mvc.ProblemDetails` to the `Microsoft.AspNetCore.Http.Extensions` assembly.

```diff
+ namespace Microsoft.AspNetCore.Http.Extensions
+ {
+     public class HttpValidationProblemDetails
+     {
+         public HttpValidationProblemDetails();
+         public HttpValidationProblemDetails(IDictionary<string, string[]> errors);
+         public IDictionary<string, string[]> Errors { get; }
+     }
+ }

namespace Microsoft.AspNetCore.Mvc
{
-   public class ValidationProblemDetails : ProblemDetails
+   public class ValidationProblemDetails : HttpValidationProblemDetails
    {
    }
}
```

### PR DESCRIPTION
**Pull request by pranavkm**
* Type-forward ProblemDetails to Http.Extensions
* Introduce HttpValidationProblemDetails to support validation details
* Remove IResult from StatusCodeResult

Fixes https://github.com/dotnet/aspnetcore/issues/34169

### CODE REVIEW

File: src/Http/Http.Extensions/src/HttpValidationProblemDetails.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Microsoft.AspNetCore.Http.Extensions
+{
+    /// <summary>
+    /// A <see cref="ProblemDetails"/> for validation errors.
+    /// </summary>
+    [JsonConverter(typeof(HttpValidationProblemDetailsJsonConverter))]
+    public class HttpValidationProblemDetails : ProblemDetails
```
Comment (davidfowl) [inline src/Http/Http.Extensions/src/HttpValidationProblemDetails.cs:15]: sealed?

Comment (pranavkm) [reply src/Http/Http.Extensions/src/HttpValidationProblemDetails.cs:15]:
```diff
namespace Microsoft.AspNetCore.Mvc
{
-   public class ValidationProblemDetails : ProblemDetails
+   public class ValidationProblemDetails : HttpValidationProblemDetails
    {
    }
}
```

Comment (pranavkm) [reply src/Http/Http.Extensions/src/HttpValidationProblemDetails.cs:15]: I'm open to not doing this - it just felt like a good way to help move the feature over to Http.Extensions.

**Review by halter73 — APPROVED** [review]

File: src/Http/Http.Extensions/src/HttpValidationProblemDetails.cs (hunk 1 above)
Comment (davidfowl) [reply src/Http/Http.Extensions/src/HttpValidationProblemDetails.cs:15]: I was originally thinking we would have ProblemDetails derive from something in the core ProblemDetailsBase.

**Pull request merged**